### PR TITLE
filebeat: Fix flaky test case on macOS

### PR DESCRIPTION
**Pull request by VihasMakwana**
## Proposed commit message

This PR fixes following test cases which are flaky for macOS:

- tests/system/test_harvester.py::Test::test_debug_reader
- tests/system/test_crawler.py::Test::test_tail_files
- tests/system/test_crawler.py::Test::test_encodings
- tests/system/test_registrar.py::Test::test_restart_state_reset_ttl_no_clean_inactive
- tests/system/test_registrar.py::Test::test_restart_state_reset
- tests/system/test_registrar.py::Test::test_restart_state
- tests/system/test_registrar.py::Test::test_registry_file_update_permissions
- tests/system/test_shutdown.py::Test::test_shutdown

On MacOS, the FQDN lookup takes some time (~5 seconds) before signal handlers are set up 
  - the lingering FQDN lookup code is [here](https://github.com/elastic/beats/blob/main/libbeat/cmd/instance/beat.go#L836-L844)
To make sure that the tests are passed, we need to increase the timeout by few seconds in each of the failing cases.

You can view the full discussion https://elastic.slack.com/archives/C047NDNGUMU/p1717091056290869.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

Fixes https://github.com/elastic/beats/issues/39613

### CODE REVIEW

File: filebeat/tests/system/test_registrar.py [hunk 1]
```diff
@@ -269,7 +269,7 @@ def test_registry_file_update_permissions(self):
             max_timeout=1)
 
         # Wait a moment to make sure registry is completely written
-        time.sleep(1)
+        time.sleep(10)
```
Comment (belimawr) [inline filebeat/tests/system/test_registrar.py:272]: Do we really need to wait 10s here?

Can we replace the next assertion for something like `self.wait_until` that will keep trying until the condition is true or the timeout is reached?

That will help the tests to run faster.

File: filebeat/tests/system/test_registrar.py [hunk 2]
```diff
@@ -975,8 +975,8 @@ def test_restart_state_reset(self):
 
         self.render_config_template(
             path=os.path.abspath(self.working_dir) + "/log/*",
-            clean_inactive="10s",
-            ignore_older="5s"
+            clean_inactive="20s",
+            ignore_older="10s"
```
Comment (belimawr) [inline filebeat/tests/system/test_registrar.py:979]: If we're waiting for the FQDN to be resolved we don't need to change the input settings.

I can see `ignore_older` needing to be increased if it takes longer for Filebeat to start up and scan for the files, but I do not get the change in `clean_inactive`

**Review by belimawr — COMMENTED** [review]
A general question: Do we need those tests using the FQDN? Could we just disable it as those tests are not testing the FQDN and have the suit run faster?

**Comment by VihasMakwana** [comment]
> A general question: Do we need those tests using the FQDN? Could we just disable it as those tests are not testing the FQDN and have the suit run faster?

Hmm, I'm not sure if we can selectively disable FQDN. These integration test cases try to run the `filebeat` binary and test for output.
Also, this problem is specific to Darwin based systems.
On other systems, the `wait` calls should resolve quickly.

File: filebeat/tests/system/test_registrar.py (hunk 2 above)
Comment (VihasMakwana) [reply filebeat/tests/system/test_registrar.py:979]: Oh! I think I this was added by accident.
You're right, we only need to change `ignore_older`

**Comment by belimawr** [comment]
> > A general question: Do we need those tests using the FQDN? Could we just disable it as those tests are not testing the FQDN and have the suit run faster?
> 
> Hmm, I'm not sure if we can selectively disable FQDN. These integration test cases try to run the `filebeat` binary and test for output. Also, this problem is specific to Darwin based systems. On other systems, the `wait` calls should resolve quickly.

The Elastic-Agent has got a flag to select if we're using FQDN or not, I don't fully recall how that works for a standalone Beat, it might be possible.

But if the OS is slow to respond regardless of how we query it, then there is not much we can do.

Anyway, the PR looks good now!

**Review by belimawr — APPROVED** [review]

**Review by rdner — APPROVED** [review]

**Pull request merged**